jacodb-analysis/src/main/kotlin/org/jacodb/analysis/util/EtsTraits.kt

```diff
@@ -0,0 +1,160 @@
+import org.jacodb.analysis.util.toPathOrNull
+import org.jacodb.api.common.CommonMethodParameter
+import org.jacodb.api.common.cfg.CommonArgument
+import org.jacodb.api.common.cfg.CommonCallExpr
+import org.jacodb.api.common.cfg.CommonExpr
+import org.jacodb.api.common.cfg.CommonValue
+import org.jacodb.panda.dynamic.ets.base.EtsAnyType
+import org.jacodb.panda.dynamic.ets.base.EtsArrayAccess
+import org.jacodb.panda.dynamic.ets.base.EtsCallExpr
+import org.jacodb.panda.dynamic.ets.base.EtsCastExpr
+import org.jacodb.panda.dynamic.ets.base.EtsConstant
+import org.jacodb.panda.dynamic.ets.base.EtsEntity
+import org.jacodb.panda.dynamic.ets.base.EtsImmediate
+import org.jacodb.panda.dynamic.ets.base.EtsInstanceFieldRef
+import org.jacodb.panda.dynamic.ets.base.EtsParameterRef
+import org.jacodb.panda.dynamic.ets.base.EtsStaticFieldRef
+import org.jacodb.panda.dynamic.ets.base.EtsStmt
+import org.jacodb.panda.dynamic.ets.base.EtsThis
+import org.jacodb.panda.dynamic.ets.base.EtsValue
+import org.jacodb.panda.dynamic.ets.model.EtsMethod
+import org.jacodb.panda.dynamic.ets.model.EtsMethodImpl
+import org.jacodb.panda.dynamic.ets.model.EtsMethodParameter
+import org.jacodb.panda.dynamic.ets.utils.callExpr
+import org.jacodb.taint.configuration.ConstantValue
+import org.jacodb.analysis.util.toPath as _toPath
+import org.jacodb.analysis.util.toPathOrNull as _toPathOrNull
+import org.jacodb.panda.dynamic.ets.utils.getOperands as _getOperands
+import org.jacodb.panda.dynamic.ets.utils.getValues as _getValues
+
+interface EtsTraits : Traits<EtsMethod, EtsStmt> {
+
+    override val CommonCallExpr.callee: EtsMethod
+        get() {
+            check(this is EtsCallExpr)
+            // return cp.getMethodBySignature(method) ?: error("Method not found: $method")
+            return EtsMethodImpl(method)
+        }
+
+    override val EtsMethod.thisInstance: EtsThis
+        get() = EtsThis(EtsAnyType)
+
+    override val EtsMethod.isConstructor: Boolean
+        get() = false
+
+    override fun CommonExpr.toPathOrNull(): AccessPath? {
+        check(this is EtsEntity)
+        return this._toPathOrNull()
+    }
+
+    override fun CommonValue.toPathOrNull(): AccessPath? {
+        check(this is EtsValue)
+        return this._toPathOrNull()
+    }
+
+    override fun CommonValue.toPath(): AccessPath {
+        check(this is EtsValue)
+        return this._toPath()
+    }
+
+    override fun getArgument(param: CommonMethodParameter): EtsParameterRef {
+        check(param is EtsMethodParameter)
+        return EtsParameterRef(index = param.index, type = param.type)
+    }
+
+    override fun getArgumentsOf(method: EtsMethod): List<CommonArgument> {
+        return method.parameters.map { getArgument(it) }
+    }
+
+    override fun CommonValue.isConstant(): Boolean {
+        check(this is EtsEntity)
+        return this is EtsConstant
+    }
+
+    override fun CommonValue.eqConstant(constant: ConstantValue): Boolean {
+        TODO("Not yet implemented")
```

Comment:
Because someone was too lazy to think about the proper implementation.

Yes, we need it, since we have ConstantEq condition in taint configuration. We do not actually use it in taint configs for Ets, but that's another story.